pallets/messages/src/lib.rs

```diff
@@ -365,7 +370,17 @@ impl<T: Config> Pallet<T> {
 				list.len().try_into().map_err(|_| Error::<T>::TypeConversionOverflow)?;
 			for i in from_index..list_size {
 				let m = list[i as usize].clone();
-				response.content.push(m.map_to_response(block_number));
+
+				let payload: Vec<u8>;
+				let payload_length: u32;
+				(payload, payload_length) = match payload_location {
```

Comment:
```suggestion
				let (payload: Vec<u8>, payload_length: u32) = match payload_location {
```

Reply:
I had been trying to do that, not sure why but Rust doesn't like the type constraints in the tuple

Reply:
Ooo this worked! `let (payload, payload_length): OffchainPayloadType = match ...`